docs/commits_and_prs.md

```diff
@@ -0,0 +1,105 @@
+* They improve the quality of your work by encouraging clearly organized thinking, modular code, and purity of purpose.
+
+Make commits atomic: CI should be passing after each commit to enable tools
+like `git revert` and `git bisect` to be used effectively and to make history
+easier to browse and search by avoiding "fix", "typo" and "test" commits. Run
+`make sure` locally to make sure that CI will pass.
+
+<details><summary><b>When should you include more than one commit in a PR?</b></summary>
+If a few closely related commits would be easier to review and deploy together
+then you might want to include them in the same PR.
+If commits can be decoupled into independently-reviewable, smaller PRs then it
+might be better to split them up.
+</details>
+
+Write helpful commit messages, PR titles, and PR bodies
+-------------------------------------------------------
+
+Good commit messages make tools like `git blame`, `git revert` and `git log`
+come to life. Good titles and bodies make PRs easier to review by explaining
+your intent, reasoning and context rather than making your reviewer figure it
+out.
+
+[Every line of code is always documented](https://mislav.net/2014/02/hidden-documentation/):
+write to explain your change to the whole team, not just the individual who'll review it.
+Enable future code archaeology: make it possible to understand and revisit your
+change months or years from now.
+
+As with breaking changes into atomic commits and small PRs, writing good commit
+and PR messages improves the quality of your work: it's a self-review that helps
+you clearly organize your thoughts and understand what you've done and whether
+it really solves the problem.
+
+#### What should go in commit and PR messages?
+
+A commit's diff shows **how** the change was made: what lines of code were
+changed. It's up to the commit message to document **what you intended** the
+commit to achieve, **why** you're making this change, and any **context**
+or background knowledge necessary to understand the change.
+[My favourite Git commit](https://dhwthompson.com/2019/my-favourite-git-commit) (written by a former Hypothesis developer)
+and [Telling stories through your commits](https://blog.mocoso.co.uk/talks/2015/01/12/telling-stories-through-your-commits/)
+have some great tips.
+
+The first line of a commit should summarize the intent in less than 50 chars
+because it works well with Git and GitHub which treat the first line as a title
+and use it throughout their interfaces.
+Use the [imperative mood](https://en.wikipedia.org/wiki/Imperative_mood) because it's short and clear.[^1]
+
+For single-commit PRs GitHub pre-fills the title and body from the commit
+message. You might add some more details yourself: screenshots, testing
+instructions, review notes. Multi-commit PR titles can follow the same style as
+for single commits[^2], the PR title and body can document the PR as a whole
+while commit messages can add detail to individual commits.
```

Comment:
I've had a go at editing this. I'm trying to say something about what should go in the PR title and body versus in the commit message. For single-commit PRs I think the PR title and body are the same as the commit message (and GitHub even suggests that by pre-filling them) but you might want to add some extra to the PR body that's not in the commit message. For multi-commit PRs I'm trying to say that the PR title and body should describe the PR as a whole rather than, for example, copy-pasting each one of the commit messages.